src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -16,170 +16,261 @@
-    Pair.pair("fprintf", 1),
-    Pair.pair("fscanf", 1),
-    Pair.pair("scanf", 0),
-    Pair.pair("sscanf", 1),
-    Pair.pair("fatalf", 0),
-    Pair.pair("panicf", 0),
-    Pair.pair("logf", 0),
-    Pair.pair("skipf", 0));
-
-  private static int getPlaceholderPosition(@NotNull GoFunctionOrMethodDeclaration function) {
-    Integer position = FORMATTING_FUNCTIONS.get(StringUtil.toLowerCase(function.getName()));
-    if (position != null) {
-      String importPath = function.getContainingFile().getImportPath(false);
-      if ("fmt".equals(importPath) || "log".equals(importPath) || TESTING_PATH.equals(importPath)) {
-        return position;
-      }
-    }
-    return -1;
-  }
 
-  private static int getPlaceholders(@Nullable GoExpression argument) {
-    if (argument == null) return -1;
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitCallExpr(@NotNull GoCallExpr o) {
+        PsiReference psiReference = o.getExpression().getReference();
+        PsiElement resolved = psiReference != null ? psiReference.resolve() : null;
+        if (!(resolved instanceof GoFunctionOrMethodDeclaration)) return;
 
-    String placeholderText = resolve(argument);
-    if (placeholderText == null) return -1;
+        String functionName = StringUtil.toLowerCase(((GoFunctionOrMethodDeclaration)resolved).getName());
+        if (functionName == null) return;
 
-    Matcher placeholders = PLACEHOLDER_PATTERN.matcher(placeholderText);
-    Matcher countPlaceholders;
+        if (GoPlaceholderChecker.isFormattingFunction(functionName)) {
+          checkPrintf(holder, o, (GoFunctionOrMethodDeclaration)resolved);
+        }
+        else if (GoPlaceholderChecker.isPrintingFunction(functionName)) {
+          checkPrint(holder, o, (GoFunctionOrMethodDeclaration)resolved);
+        }
+      }
+    };
+  }
 
-    int indexCount = 0;
-    int maxIndexCount = 0;
+  private static void checkPrint(
+    @NotNull ProblemsHolder holder,
+    @NotNull GoCallExpr callExpr,
+    @NotNull GoFunctionOrMethodDeclaration declaration) {
 
-    while (placeholders.find()) {
-      indexCount++;
+    List<GoExpression> arguments = callExpr.getArgumentList().getExpressionList();
+    if (arguments.isEmpty()) return;
 
-      countPlaceholders = INDEXED_PLACEHOLDER_PATTERN.matcher(placeholders.group());
-      while (countPlaceholders.find()) {
-        int index = StringUtil.parseInt(countPlaceholders.group().replaceAll("\\[|\\]", ""), -1);
-        if (index > 0 && index != indexCount) {
-          indexCount = index;
+    GoExpression firstArg = arguments.get(0);
+    if (GoTypeUtil.isString(firstArg.getGoType(null))) {
+      String firstArgText = GoPlaceholderChecker.resolve(firstArg);
+      if (firstArgText != null) {
+        if (GoPlaceholderChecker.hasPlaceholder(firstArgText)) {
```

Comment:
Unfortunately yes. At least that's the current behavior of `go vet`. Maybe we should make this smarter and call for this if the string actually contains a formatting directive?